File: lib/java/architecture/Validator.java
package architecture;



import java.util.Map;

import mmiLib.Exception;
import mmiLib.MessageType;

public class Validator {
	public static String validate(Map<String, String> returnedValues, String requestId) {
		System.out.println("[Validator][validate] - Started ");
		String message = "";
		String type = "";
		if(returnedValues.get("type") == null) {
			message += Exception.TYPE_EMPTY + "\n";
		} else {
			type = returnedValues.get("type");
			message += "messageType: " + type + "\n";
			
			// ******************** Message base properties ********************
			if (returnedValues.get("id") == null)
				message += Exception.REQUEST_ID_EMPTY + "\n";
			else {
				if (!returnedValues.get("id").equalsIgnoreCase(requestId))
					message += "RequestId not matched: received: "+ returnedValues.get("id")
						+ " current: " + requestId + "\n";
					System.out.println("[Validator][validate] - returnedId: " + returnedValues.get("id") + 
							" currentId: " + requestId);
			}
				
			
			if (returnedValues.get("source") == null)
				message += Exception.SOURCE_EMPTY + "\n";			
			
			if (returnedValues.get("target") == null)
				message += Exception.TARGET_EMPTY + "\n";
			
			if (returnedValues.get("context") == null)
				message += Exception.CONTEXT_EMPTY + "\n";
			
			// ******************** Message response properties ********************
			if (returnedValues.get("deliveryMode") == null)
				message += Exception.DELIVERY_MODE_EMPTY + "\n";
			
			if (returnedValues.get("status") == null)
				message += Exception.STATUS_EMPTY + "\n";
			
			// ******************** Specific properties ********************
			if (type.equals(MessageType.STATUS.status())) {
				if (returnedValues.get("automaticUpdate") == null)
					message += Exception.AUTOMATIC_UPDATE_EMPTY + "\n";
			} else if (type.equals(MessageType.NEW_CONTEXT.status())) {
				// No more properties
			} else if (type.equals(MessageType.PREPARE.status())) {
				// No more properties
			} else if (type.equals(MessageType.START.status())) {
				// No more properties
			} else {
				// @Todo : default
			}
			
			
			// Don't have errors
			if (message.equals("messageType: " + type + "\n") )
				message = "OK";
			
		}
		System.out.println("[Validator][validate] - Ended ");
		return message;
	}
}

File: lib/js/Architecture_Lib/Validator.js
var Validator = {
	validate : function(serviceResponse, requestId) { 
		
		console.log("[Validator]{validate} =============================== MMI EVENT TYPE RECEIVED: "+getEventType(serviceResponse.type)+" code "+serviceResponse.type);
		var message = "";
		var type = "";
		if(serviceResponse.type == null) {
			message += exception.TYPE_EMPTY + "\n";
		} else {
			type = serviceResponse.type;
			message += "messageType: " + type + "\n";
			console.log("[Validator]{validate} Verifies Message base properties "); 
			if(serviceResponse.id == null)
				message += exception.REQUEST_ID_EMPTY + "\n";
			else {
				if(serviceResponse.id != requestId)
					message += "RequestId not matched, requestId: " + requestId 
					+ " received: " + serviceResponse.id +  "\n";
			}

			if(serviceResponse.source == null)
				message += exception.SOURCE_EMPTY + "\n";

			if(serviceResponse.target == null)
				message += exception.TARGET_EMPTY + "\n";

			if(serviceResponse.context == null)
				message += exception.CONTEXT_EMPTY + "\n"
			
			console.log("[Validator]{validate} Verifies Message response properties "); 
			if(serviceResponse.deliveryMode == null)
				message += exception.DELIVERY_MODE_EMPTY + "\n";

			// if(serviceResponse.delivery == null)
				// message += exception.IS_DELIVERED_EMPTY + "\n";

			if(serviceResponse.status == null)
				message += exception.STATUS_EMPTY + "\n";
			else {
					var currentStatus = serviceResponse.status;
				}
			
			console.log("[Validator]{validate} Verifies Specific properties "); 
			switch (type) {
				case messageType.STATUS :
					if(serviceResponse.automaticUpdate == null)
						message += exception.AUTOMATIC_UPDATE_EMPTY + "\n";

					break;

				case messageType.NEW_CONTEXT :
					// No more properties
					break;

				case messageType.PREPARE :
					// No more properties

					break;

				case messageType.START :
				// No more properties

				default :
					break;
			}

			// Don't have errors
			if(message == "messageType: " + type + "\n")
				message = "OK";
		}

		// Return validation result
		if(message == "")
			message = "OK";
		return message; 
	}
}
console.log("Observer_Lib loaded");